lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -39,6 +39,16 @@ def default_priority
     self.priority = MiqAeDomain.highest_priority(tenant) + 1 unless priority
   end
 
+  def lock
+    self.system = true
+    save
```

Comment:
Additonally, if you do decide to go the `locked=` route, it's bad form to save in a setter.  The caller can make that decision by doing `update_attributes!(:locked, true)` which is more natural of an API.